cmd/dra-example-controller/claimparametersgen.go

```diff
@@ -0,0 +1,229 @@
+
+	// Set up handler for events
+	_, err = gpuClaimParametersInformer.AddEventHandler(newGpuClaimParametersHandler(ctx, config.clientSets.Core, dynamicClient))
+	if err != nil {
+		return fmt.Errorf("error adding event handler: %w", err)
+	}
+
+	// Start informer
+	go gpuClaimParametersInformer.Run(ctx.Done())
+
+	return nil
+}
+
+func newGpuClaimParametersInformer(ctx context.Context, dynamicClient dynamic.Interface) cache.SharedIndexInformer {
+	// Set up shared index informer for GpuClaimParameters objects
+	gvr := schema.GroupVersionResource{
+		Group:    gpucrd.GroupName,
+		Version:  gpucrd.Version,
+		Resource: strings.ToLower(gpucrd.GpuClaimParametersKind),
+	}
+
+	informer := cache.NewSharedIndexInformer(
+		&cache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				return dynamicClient.Resource(gvr).List(ctx, metav1.ListOptions{})
+			},
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				return dynamicClient.Resource(gvr).Watch(ctx, metav1.ListOptions{})
+			},
+		},
+		&unstructured.Unstructured{},
+		0, // resyncPeriod
+		cache.Indexers{},
+	)
+
+	return informer
+}
+
+func newGpuClaimParametersHandler(ctx context.Context, clientset kubernetes.Interface, dynamicClient dynamic.Interface) cache.ResourceEventHandler {
+	return cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj any) {
+			unstructured, ok := obj.(*unstructured.Unstructured)
+			if !ok {
+				klog.Errorf("Error converting object to *unstructured.Unstructured: %v", obj)
+			}
+
+			var gpuClaimParameters gpucrd.GpuClaimParameters
+			err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructured.Object, &gpuClaimParameters)
+			if err != nil {
+				klog.Errorf("Error converting *unstructured.Unstructured to GpuClaimParameters: %v", err)
+				return
+			}
+
+			if err := createOrUpdateResourceClaimParameters(ctx, clientset, &gpuClaimParameters); err != nil {
+				klog.Errorf("Error creating ResourceClaimParameters: %v", err)
+				return
+			}
+		},
+		UpdateFunc: func(oldObj any, newObj any) {
+			unstructured, ok := newObj.(*unstructured.Unstructured)
+			if !ok {
+				klog.Errorf("Error converting object to *unstructured.Unstructured: %v", newObj)
+			}
+
+			var gpuClaimParameters gpucrd.GpuClaimParameters
+			err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructured.Object, &gpuClaimParameters)
+			if err != nil {
+				klog.Errorf("Error converting *unstructured.Unstructured to GpuClaimParameters: %v", err)
+				return
+			}
+
+			if err := createOrUpdateResourceClaimParameters(ctx, clientset, &gpuClaimParameters); err != nil {
+				klog.Errorf("Error updating ResourceClaimParameters: %v", err)
+				return
+			}
+		},
+	}
+}
+
+func newResourceClaimParametersFromGpuClaimParameters(gpuClaimParameters *gpucrd.GpuClaimParameters) (*resourceapi.ResourceClaimParameters, error) {
+	namespace := gpuClaimParameters.Namespace
```

Comment:
This looks redundand as the `namespace` variable is used only once below.

Reply:
Removed in latest